libavcodec/qsvenc.c

```diff
@@ -164,6 +164,13 @@ static const struct {
 #endif
 };
 
+#define UPDATE_PARAM(a, b)  \
+    if ((a) != (b))         \
+    {                       \
+        a = b;              \
+        updated = 1;     \
+    }
+
```

Comment:
After extending `UPDATE_PARAM(a, b);`,  there is `;` after `}'.  I prefer not to place a `;` after `}`. 

How about to use `do {} while(0)` in this macro ?

Reply:
Ah, that is why there are macro definition in ffmpeg using "do {} while(0)" structure, which confused me.
Thanks for your advice. I will update the PR.

Reply:
Done